eng/scripts/Cadl-Project-Sync.ps1

```diff
@@ -0,0 +1,101 @@
+[CmdletBinding()]
+param (
+    [Parameter(Position=0)]
+    [ValidateNotNullOrEmpty()]
+    [string] $ProjectDirectory
+)
+
+$ErrorActionPreference = "Stop"
+
+function AddSparseCheckoutPath([string]$subDirectory, [string]$file) {
+    if (!(Test-Path $file) -or !((Get-Content $file).Contains($subDirectory))) {
+        Write-Output $subDirectory >> .git/info/sparse-checkout
+    }
+}
+
+function NpmInstallForProject([string]$workingDirectory) {
+    Push-Location $workingDirectory
```

Comment:
Whenever pushing locations, right below it use:

```powershell
trap { Pop-Location }
```

This is an old school try..finally mechanism in PowerShell that makes sure, if errors terminate the current scope, code runs. If you don't and errors occur, your script will be stuck in this directory and bad things can happen.